Source/Urho3D/UI/UIComponent.cpp

```diff
@@ -182,25 +193,51 @@ bool UIComponent::ScreenToUIPosition(IntVector2 screenPos, IntVector2& result)
     octree->Raycast(query);
 
     if (queryResultVector.Empty())
-        return false;
+        return result;
 
     for (unsigned i = 0; i < queryResultVector.Size(); i++)
     {
         RayQueryResult& queryResult = queryResultVector[i];
-        if (queryResult.drawable_ != model_)
+        if (queryResult.drawable_ != component_->GetModel())
         {
             // ignore billboard sets by default
             if (queryResult.drawable_->GetTypeInfo()->IsTypeOf(BillboardSet::GetTypeStatic()))
                 continue;
-            return false;
+            return result;
         }
 
         Vector2& uv = queryResult.textureUV_;
-        result = IntVector2(static_cast<int>(uv.x_ * rootElement_->GetWidth()),
-                            static_cast<int>(uv.y_ * rootElement_->GetHeight()));
-        return true;
+        result = IntVector2(static_cast<int>(uv.x_ * component_->GetRoot()->GetWidth()),
```

Comment:
But `component_->GetRoot()` is always `this`, am I right?

Reply:
`component_->GetRoot()` returns `UIElement` which acts as a root element. This is same `UIElement` that is passed to `UI::SetRenderTexture()`